src/ripple/app/tests/Offer.test.cpp

```diff
@@ -235,12 +253,393 @@ class Offer_test : public beast::unit_test::suite
             env.require (balance (carol, USD2 (50)));
         }
     }
+
+    void
+    testInsufficientReserve ()
+    {
+        testcase ("Insufficient Reserve");
+
+        // If an account places an offer and its balance
+        // *before* the transaction began isn't high enough
+        // to meet the reserve *after* the transaction runs,
+        // then no offer should go on the books but if the
+        // offer partially or fully crossed the tx succeeds.
```

Comment:
We detect the "fill or kill" case earlier, so such an offer would have simply returned `tesSUCCESS`,  and done nothing but charge a fee.